### PR TITLE
Make `gears` cross-platform and some other fancy magic

### PR DESCRIPTION
**Pull request by natsukagami**
Most tests on native are running but not all (I ignored `ChannelBehavior` tests but they are still dead ;w;)

- Make gears cross-platform
- Add organization
- Simple fork-join executor
- Add ForkJoin support
- Implement Timer and some efforts to make tests run
- Move scala to 3.3.1
- Get timer to run
- Get other tests to run on all platforms
- Split test framework to support a custom version of munit-native
- Use `FiniteDuration` in schedule
- Make timer tests pass
- Introduce cleanup state for RunnableFuture for... cleanup
- Don't wake up prematurely

### CODE REVIEW

File: shared/src/main/scala/async/Async.scala [hunk 1]
```diff
@@ -47,7 +46,12 @@ object Async:
    *  The thread will suspend when the computation waits.
    */
   def blocking[T](body: Async ?=> T)(using support: AsyncSupport, scheduler: support.Scheduler): T =
-    body(using Blocking(CompletionGroup.Unlinked))
+    val async = Blocking(CompletionGroup.Unlinked)
+    val group = new CompletionGroup
```
Comment (m8nmueller) [inline shared/src/main/scala/async/Async.scala]: ~I suggest to just use Async.group~ I will implement this on my thread.

File: shared/src/main/scala/async/futures.scala [hunk 2]
```diff
@@ -137,7 +141,9 @@ object Future:
             cancellable.suspension = k
             cancellable.listener = listener
             src.onComplete(listener)
-            innerGroup.add(cancellable) // may resume + remove listener immediately
+            if !cleanup then // during cleanup we already cancelled the innerGroup
```
Comment (m8nmueller) [inline shared/src/main/scala/async/futures.scala]: TODO to be removed after my fix

File: jvm/src/main/scala/async/VThreadSupport.scala [hunk 3]
```diff
@@ -2,19 +2,19 @@ package gears.async
...
 
 object VThreadScheduler extends Scheduler:
   override def execute(body: Runnable): Unit = Thread.startVirtualThread(body)
 
-  override def schedule(delay: Duration, body: Runnable): Cancellable =
+  override def schedule(delay: FiniteDuration, body: Runnable): Cancellable =
     val th = Thread.startVirtualThread: () =>
-      Thread.sleep(delay)
-      body.run()
-    () => th.interrupt() // TODO this may interrupt the body after sleeping
```
Comment (m8nmueller) [inline jvm/src/main/scala/async/VThreadSupport.scala:17]: Is there a subject-related reason for removing this note, i.e., did you fix it?

Comment (natsukagami) [reply jvm/src/main/scala/async/VThreadSupport.scala:17]: Yeah, I basically just end the sleeping thread and put the `body` into the scheduler; that way the body is run in another v-thread and not get interrupted.

**Pull request closed without merging**